pip/basecommand.py

```diff
@@ -104,6 +114,9 @@ def parse_args(self, args):
         # factored out for testability
         return self.parser.parse_args(args)
 
+    def running_outside_virtualenv(self):
```

Comment:
Alex, `is_in_virtualenv` would require negative logic in the `if` statement below, which is not ideal.